src/Tasks/Microsoft.Common.CurrentVersion.targets

```
@@ -614,8 +614,9 @@ Copyright (C) Microsoft Corporation. All rights reserved.
   </PropertyGroup>
 
   <!-- Users familiar with how some other repos work try to use NoWarn with MSBuild in place of MSBuildWarningsAsMessages. -->
-  <PropertyGroup Condition="'$(MSBuildWarningsAsMessages)'=='' And $([MSBuild]::AreFeaturesEnabled('16.8'))">
-    <MSBuildWarningsAsMessages>$(NoWarn)</MSBuildWarningsAsMessages>
+  <PropertyGroup Condition="$([MSBuild]::AreFeaturesEnabled('16.8'))">
+    <MSBuildWarningsAsMessages Condition="'$(MSBuildWarningsAsMessages)'==''">$(NoWarn)</MSBuildWarningsAsMessages>
+    <MSBuildWarningsAsErrors Condition="'$(MSBuildWarningsAsErrors)'==''">$(WarnAsError)</MSBuildWarningsAsErrors>
```

```suggestion
    <MSBuildWarningsAsErrors Condition="'$(MSBuildWarningsAsErrors)'==''">$(WarningsAsErrors)</MSBuildWarningsAsErrors>
```

I think it should be `WarningsAsErrors` - see for example https://github.com/dotnet/msbuild/blob/e5218a90cf743c8bd5f489c36e72643d8a8c6c61/src/Tasks/Microsoft.CSharp.CurrentVersion.targets#L302

---

[The issue](https://github.com/dotnet/msbuild/issues/4421#issuecomment-573915116) said WarnAsError—I think this is about matching what Roslyn does rather than what other parts of MSBuild does, since this is an MSBuild weird syntax issue.

---

The issue was incorrect: https://github.com/dotnet/roslyn/blob/c64d1e3a7b3e2b6d5a71ca4ceb5848bcd79a0970/src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets#L144